### ISSUE TITLE
400 Client Error: Bad Request Error Message: Resource not found for the segment 'sendMail'.

### ISSUE THREAD

**Issue by technqvi** [evyd13/django-o365mail#8]
Hi

I try to setup up the package as your advice as a prior question.
this link  [https://github.com/evyd13/django-o365mail/issues/7](https://github.com/evyd13/django-o365mail/issues/7)

**_400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/sendMail | Error Message: Resource not found for the segment 'sendMail'._**

I click to https://graph.microsoft.com/v1.0/sendMail   in order to dig deeper into additional error

{"error":{"code":"InvalidAuthenticationToken","message":"Access token is empty.","innerError":{"date":"2022-11-24T13:45:25","request-id":"4b533efc-d806-4829-a506-ba03599ebd69","client-request-id":"4b533efc-d806-4829-a506-ba03599ebd69"}}}

<img width="652" alt="image" src="https://user-images.githubusercontent.com/38780060/203799635-78ceef07-4218-4070-ae95-a379e8e1e19b.png">

Below are my code and configuration and package installation.

![image](https://user-images.githubusercontent.com/38780060/203774365-8ed6df17-a851-4282-9824-201c30ccc4b2.png)

settings.py
``` 
EMAIL_BACKEND = 'django_o365mail.EmailBackend'
O365_MAIL_CLIENT_ID ='xxxxxxxxxxxxxxx'
O365_MAIL_CLIENT_SECRET ='yyyyzzzyyyyy'
O365_MAIL_TENANT_ID ='zzzzzzzzz'
O365_ACTUALLY_SEND_IN_DEBUG=True
 ```

views.py

``` 
from django.core.mail import BadHeaderError, send_mail
from django.http import HttpResponse
def send_mail_office365(request):
    try:
        send_mail(
            subject='SmartApp - Test Sending Email',
            message='Hi, ok',
            from_email='smartapp-service@xxxxxgroup.com',
            recipient_list=['Taned.Sa@xxxxx.com'])
    except BadHeaderError:
        return HttpResponse('Invalid header found.')
    return render(request,
                  'app/pm_doc_gen_pdf.html',
                  {})
``` 

The last picture is what the azure administrator of my company set up.
![image](https://user-images.githubusercontent.com/38780060/203774312-834c2cb5-aace-497e-8cda-c9284cf3ab39.png)

Thank you for your advice

**Comment by dejaeghered** [comment]
The error message is related to the Office 365 and not to this package.
I assume somewhere your permissions or authentication is not correct.
Check the graphql documentation from Microsoft on this specific error

Also dont forget to set the mailbox!
O365_MAIL_MAILBOX_KWARGS = {'resource': 'o365mailbox@domain.com'}

**Comment by technqvi** [comment]
> Hi
> 
> I try to setup up the package as your advice as a prior question. this link #7
> 
> **_400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/sendMail | Error Message: Resource not found for the segment 'sendMail'._**
> 
> I click to https://graph.microsoft.com/v1.0/sendMail in order to dig deeper into additional error
> 
> {"error":{"code":"InvalidAuthenticationToken","message":"Access token is empty.","innerError":{"date":"2022-11-24T13:45:25","request-id":"4b533efc-d806-4829-a506-ba03599ebd69","client-request-id":"4b533efc-d806-4829-a506-ba03599ebd69"}}}
> 
> <img alt="image" width="652" src="https://user-images.githubusercontent.com/38780060/203799635-78ceef07-4218-4070-ae95-a379e8e1e19b.png">
> 
> Below are my code and configuration and package installation.
I got it done.
One thing I missed is to add O365_MAIL_MAILBOX_KWARGS = {'resource': 'aaaaa@xxxxxgroup.com'} on settings.py

Thank you for supporting me.

**Comment by technqvi** [comment]
Ok

**Comment by ijay99** [comment]
I think the need to add the O365_MAIL_MAILBOX_KWARGS should be added to the documentation as it is not obvious that this is required to someone using the package for the first time. I had to also add to the Django app settings.py the e-mail for the DEFAULT_FROM_EMAIL and the SERVER_MAIL setting using the format DEFAULT_FROM_EMAIL='your email here'. Without this I was getting an error that the user account which submitted the request didn't have the right to send the email on behalf of the specified sending account. The e-mail in these two settings is the same e-mail I specified in the O365_MAIL_MAILBOX_KWARGS.

**Comment by xcojonny** [comment]
hej @technqvi ,

can you tell me your steps you've performed? I'm struggling a lot with the usage of this library.

I used the same settings as you do and the tips from @ijay99. The issue I have is 
```
403 Client Error: Forbidden for url: https://graph.microsoft.com/v1.0/users/user@tenant.com/sendMail | Error Message: Access is denied. Check credentials and try again.
```

Any Help on this would be really great.

**Comment by datengraben** [comment]
Yes please add something to the README, the default config, which does not include the relevant `resource`-key in O365_MAIL_MAILBOX_KWARGS, also did not work for me.

I have a multi-tenant app token/secret and the same IAM/Permissions configured as above and it only works, if I add the O365_MAIL_MAILBOX_KWARGS variable with the `resource` key with the value of the `from`-email adress to the active django settings.

**Comment by agdiaz** [comment]
Hello everyone, thanks for this issue. It solved my 400/Auth error when using the library. By adding the `O365_MAIL_MAILBOX_KWARGS` setting with my sender's email address, it started to work properly.
